### PR TITLE
update docs for recent changes to mkdocs-material

### PR DESCRIPTION
**Pull request by jrappen**

### CODE REVIEW

File: docs/mkdocs.yml [hunk 1]
```diff
@@ -1,58 +1,94 @@
...
+edit_uri: blob/main/docs/src/markdown
+
+docs_dir: 'src/markdown'
 
 plugins:
-  - search
-  - redirects:
-      redirect_maps:
-        'guides/cplusplus.md': 'https://github.com/sublimelsp/LSP-clangd'
-        'guides/client_configuration.md': 'client_configuration.md'
```
Comment (rchl) [inline docs/mkdocs.yml:12]: Can we keep this? It's for handling old links that might exist in various places.

File: docs/README.md [hunk 2]
```diff
@@ -1,33 +0,0 @@
...
-pip install mkdocs mkdocs-material mkdocs-redirects
-```
-
-For full documentation visit [mkdocs.org](https://www.mkdocs.org).
-
-## Commands
-
-* `mkdocs serve` - Start the live-reloading docs server.
-* `mkdocs build` - Build the documentation site.
-* `mkdocs -h` - Print help message and exit.
```
Comment (rchl) [inline docs/README.md:18]: I tend to forget how to get things running locally. Can we keep it in some form (maybe needs update regarding installing dependencies)?

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -4,17 +4,43 @@ on:
...
-        with:
-          python-version: 3.x
-      - run: pip install mkdocs-material mkdocs-redirects
-      - run: |
+
+      - name: Git config
+        run: |
+          git config --global user.email '41898282+github-actions[bot]@users.noreply.github.com'
+          git config --global user.name 'github-actions[bot]'
+          git config --global init.defaultBranch main
```
Comment (rchl) [inline .github/workflows/docs.yml:19]: Is this necessary?

File: docs/src/CNAME [hunk 4]
```diff
@@ -1 +0,0 @@
-lsp.sublimetext.io
```
Comment (predragnikolic) [inline docs/src/CNAME:1]: This file is necessary.

See
https://github.com/sublimelsp/LSP/pull/1616#issue-836406285

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -40,3 +40,32 @@ Before you submit your pull request, please review the following:
...
+- <https://github.com/mkdocs/mkdocs-redirects>
+
+Quick build via:
+
+```bash
+python3 -m pip install --upgrade pip setuptools build
+python3 -m pip install -r ./docs/src/requirements.txt
+```
+
+Docs layout:
```
Comment (rchl) [inline CONTRIBUTING.md:59]: Missing info about how to start local dev server. It was included in the old readme.

File: docs/src/requirements.txt [hunk 6]
```diff
@@ -0,0 +1,3 @@
+mkdocs
+mkdocs-material
+mkdocs-redirects
```
Comment (rchl) [inline docs/src/requirements.txt:3]: Missing some requirements here. For example the social plugin.

Comment (rchl) [reply docs/src/requirements.txt:3]: It's not about workflow but about change in `docs/mkdocs.yml`. The social plugin was not used before.

File: .github/workflows/docs.yml [hunk 7]
```diff
@@ -17,6 +17,10 @@ jobs:
 
       # Python 3 is installed by default
 
+      - name: "Install dependencies for mkdocs-material's social plugin"
+        run: |
+          apt-get install libcairo2-dev libfreetype6-dev libffi-dev libjpeg-dev libpng-dev libz-dev
```
Comment (rchl) [inline .github/workflows/docs.yml:22]: If such heavy dependencies are needed then I'm not sure it's worth including that plugin.

I don't know how to get things running locally on macOS for example and I don't really feel like figuring it out and even, if knew, I'm not sure I would want to install a ton of dev dependencies.

Comment (rchl) [reply .github/workflows/docs.yml]: Because I didn't that the social plugin will require extra system dependencies until you've added it to requirements.txt and after I've tested it.

**Comment by rchl** [comment]
```
❯ mkdocs serve                                                                                                                                                                                            
INFO     -  Building documentation...
INFO     -  Cleaning site directory
WARNING  -  Documentation file 'features.md' contains a link to '../assets/images/code-lens-annotation.png' which is not found in the documentation files.
WARNING  -  Documentation file 'features.md' contains a link to '../assets/images/code-lens-phantom.png' which is not found in the documentation files.
WARNING  -  Documentation file 'features.md' contains a link to '../assets/images/inlay-hints.png' which is not found in the documentation files.
WARNING  -  Documentation file 'features.md' contains a link to 'concepts.md' which is not found in the documentation files.
INFO     -  Documentation built in 0.32 seconds
INFO     -  [22:26:20] Watching paths for changes: 'src/markdown', 'mkdocs.yml'
INFO     -  [22:26:20] Serving on http://127.0.0.1:8000/LSP/
INFO     -  [22:26:30] Browser connected: http://127.0.0.1:8000/LSP/
INFO     -  [22:26:31] Browser connected: http://127.0.0.1:8000/LSP/
```

Screenshot 2022-11-24 at 22 28 47

**Comment by rchl** [comment]
> That `concepts.md` file doesn't exist currently in `main` nor is there a redirect. What do you wanna do with that?

You could fix the link. It's suppose to point to the previous paragraph.

**Comment by rchl** [comment]
I'm still getting warnings:
```
WARNING  -  Documentation file 'features.md' contains a link to '../assets/images/code-lens-annotation.png' which is not found in the documentation files.
WARNING  -  Documentation file 'features.md' contains a link to '../assets/images/code-lens-phantom.png' which is not found in the documentation files.
WARNING  -  Documentation file 'features.md' contains a link to '../assets/images/inlay-hints.png' which is not found in the documentation files.
```

**Comment by rchl** [comment]
> Should be fixed now, I confused source structure with structure of generated docs.

Still not working:


```
WARNING  -  Documentation file 'features.md' contains a link to 'assets/images/code-lens-annotation.png' which is not found in the documentation files.
WARNING  -  Documentation file 'features.md' contains a link to 'assets/images/code-lens-phantom.png' which is not found in the documentation files.
WARNING  -  Documentation file 'features.md' contains a link to 'assets/images/inlay-hints.png' which is not found in the documentation files.
```

File: docs/theme/main.html [hunk 8]
```diff
@@ -0,0 +1,11 @@
+{% extends "base.html" %}
+
+{% block announce %}
+  <a href="https://discordapp.com/channels/280102180189634562/645268178397560865">
+    Join us on
+    <span class="twemoji discord">
+      {% include ".icons/fontawesome/brands/discord.svg" %}
+    </span>
+    <strong>Discord</strong>
+  </a>
```
Comment (rchl) [inline docs/theme/main.html:10]: I think one has to use the invite link to join the server as explained in:

https://github.com/sublimelsp/LSP/blob/9d83fee1020cd70ffc46c6f5b4b16e78500fe763/README.md?plain=1#L50

so it's probably worth including the invitation link here also.

File: docs/theme/css/extra.css [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    height: px2em(18px);
+    vertical-align: text-top;
+
+    svg {
+      width: px2em(18px);
+      max-height: 100%;
+      fill: currentcolor;
+    }
+  }
+}
```
Comment (rchl) [inline docs/theme/css/extra.css:23]: This is CSS so nesting selectors like that is invalid. Are those even needed?

File: docs/theme/css/extra.css [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+.md-typeset :is(.twemoji) {
+    display: inline-flex;
+    height: px2em(18px);
+    vertical-align: text-top;
+}
+
+.md-typeset svg {
+      width: px2em(18px);
```
Comment (rchl) [inline docs/theme/css/extra.css:16]: Those `px2em` things are invalid (not sure why are those needed but if they are then probably some pre-processing is missing).

**Review by rchl — COMMENTED** [review]
The paths to image files still don't work.

**Comment by rwols** [comment]
It's amusing to see how the conversation goes. I'm sure you mean well @jrappen, and I appreciate that, but, next to the "writing code" phase, you also need to take care of the "engineering" phase. If you're not prepared to take care of that, it's usually best to make as little changes as possible. I think @rchl provides you a favor by pointing out what consequences your seemingly random changes have.

No maintainer wants to see things break because of merging pull requests. Because that means hours spent in a weekend where those hours could have gone to something more locally impactful.

**Comment by rwols** [comment]
Yes, writing code is the trivial part. Engineering is the non-trivial part.

> you know what you're doing anyway.

I'm afraid so. The question is who wants to put the time into it ;). Again, small changes result in higher probability of merging pull requests.

**Comment by rchl** [comment]
I don't think I was too picky in my comments and I think that my "suggestions" were actual issues (broken image links still remaining).

If you are submitting changes to a project then you are responsible for resolving issues and addressing comments raised by maintainers. If you are not willing to do that then don't submit pull request in the first place.  If you are expecting to throw half working changes at the maintainers and expect them to fix it for you then it's not how it's gonna work here at least. Testing the changes takes enough time already.
